RecoLocalCalo/HcalRecProducers/src/MahiGPU.cu

```diff
@@ -0,0 +1,1403 @@
+                                    float const* __restrict__ pulseMatricesM,
+                                    float const* __restrict__ pulseMatricesP,
+                                    int const* __restrict__ pulseOffsetValues,
+                                    float const* __restrict__ noiseTerms,
+                                    int8_t const* __restrict__ soiSamples,
+                                    float const* __restrict__ pedestalWidths,
+                                    float const* __restrict__ effectivePedestalWidths,
+                                    bool const useEffectivePedestals,
+                                    uint32_t const* __restrict__ idsf01HE,
+                                    uint32_t const* __restrict__ idsf5HB,
+                                    uint32_t const* __restrict__ idsf3HB,
+                                    float const* __restrict__ gainValues,
+                                    float const* __restrict__ respCorrectionValues,
+                                    uint32_t const nchannelsf01HE,
+                                    uint32_t const nchannelsf5HB,
+                                    uint32_t const nchannelsTotal,
+                                    uint32_t const offsetForHashes,
+                                    int const maxDepthHB,
+                                    int const maxDepthHE,
+                                    int const maxPhiHE,
+                                    int const firstHBRing,
+                                    int const lastHBRing,
+                                    int const firstHERing,
+                                    int const lastHERing,
+                                    int const nEtaHB,
+                                    int const nEtaHE) {
+      // can be relaxed if needed - minor updates are needed in that case!
+      static_assert(NPULSES == NSAMPLES);
+
+      // indices
+      auto const gch = threadIdx.x + blockIdx.x * blockDim.x;
+      auto const nchannelsf015 = nchannelsf01HE + nchannelsf5HB;
+      if (gch >= nchannelsTotal)
+        return;
+
+      // if chi2 is set to -9999 do not run minimization
+      if (outputChi2[gch] == -9999.f)
+        return;
+
+      // configure shared mem
+      extern __shared__ char shrmem[];
+      float* shrMatrixLFnnlsStorage =
+          reinterpret_cast<float*>(shrmem) + calo::multifit::MapSymM<float, NPULSES>::total * threadIdx.x;
+      float* shrAtAStorage = reinterpret_cast<float*>(shrmem) +
+                             calo::multifit::MapSymM<float, NPULSES>::total * (threadIdx.x + blockDim.x);
+
+      // conditions for pedestal widths
+      auto const id = gch < nchannelsf01HE
+                          ? idsf01HE[gch]
+                          : (gch < nchannelsf015 ? idsf5HB[gch - nchannelsf01HE] : idsf3HB[gch - nchannelsf015]);
+      //auto const id = gch >= nchannelsf01HE
```

Comment:
Please remove commented out lines, if not needed; or alternatively add another comment line which explains why they must be kept here